docs/troubleshooting-guide/db-performance.md

```diff
@@ -0,0 +1,93 @@
+---
+title: Troubleshooting database performance
+---
+
+# Troubleshooting database performance
+
+This guide deals with databases or data warehouses that are [connected to Metabase](../databases/connecting.md) as data sources.
+
+To fix problems with your Metabase [application database](../installation-and-operation/configuring-application-database.md), check out these troubleshooting guides:
+
+- [Running Metabase](./running.md).
+- [Running Metabase on Docker](./docker.md).
+- [Using or migrating from an H2 application database](./loading-from-h2.md)
+
+## Identifying bottlenecks
+
+1. Optional: Use Metabase's [auditing tools](../usage-and-performance-tools/audit.md) to look at your Metabase usage stats.\*
+2. Go to your database's server logs and check whether:
+   - Your tables are growing in size,
+   - More people are using Metabase to access your database,
+   - People are accessing your database more often, or
+   - Another script or application is accessing the database frequently.
+3. If specific tables are being queried a lot, try [Optimizing your table schemas](https://www.metabase.com/learn/analytics/which-data-warehouse).
+4. If your data, user base, and usage are starting to outgrow your database's resources, think about [scaling Metabase](https://www.metabase.com/learn/administration/metabase-at-scale) or [upgrading your hardware](https://www.metabase.com/learn/analytics/which-data-warehouse).
+5. If a third-party script or application is hitting your database with a lot of queries at a time:
+   - Stop your script or application, and [clear any queued queries](#clearing-queued-queries).
+   - Recommended: Add a timeout to your script, schedule the script or application to run during off-hours, or replicate your database (and point your tools there instead).
+
+\* Available on paid plans.
+
+## Resetting a database connection
+
+1. Go to **Settings** > **Admin settings** > **Databases** > your database.
+2. Click **Save changes** (without making changes) to reset Metabase's connections to your database.
```

Comment:
@mazameli, if this is a thing people need to do often, we may want to consider a "Reset connection" button in the UI.